src/api/mod.rs

```diff
@@ -115,3 +137,154 @@ fn build_fetch_query<T: Serialize>(path: &str, form: T) -> String {
+    .cookie()
+    .map_err(|_| LemmyAppError::APIError {
+      error: String::from("Cookie could not be set"),
+    })?;
+
+  wasm_cookies::cookies::get(&r, path)
+    .ok_or(LemmyAppError::APIError {
+      error: String::from("DOM cookie is None"),
+    })?
+    .map_err(|e| LemmyAppError::APIError {
+      error: e.to_string(),
+    })
+}
+
+#[cfg(not(feature = "ssr"))]
+pub async fn set_cookie_wrapper(path: &str, value: &str) -> Result<(), LemmyAppError> {
+  use leptos::window;
+  use wasm_bindgen::JsCast;
+  use wasm_cookies::CookieOptions;
+
+  let r = window()
+    .document()
+    .ok_or(LemmyAppError::APIError {
+      error: String::from("DOM document is None"),
+    })?
+    .dyn_into::<web_sys::HtmlDocument>()
+    .map_err(|_| LemmyAppError::APIError {
+      error: String::from("DOM document could not be cast"),
+    })?
+    .set_cookie(&wasm_cookies::cookies::set(path, value, &CookieOptions::default())[..])
+    .map_err(|_| LemmyAppError::APIError {
+      error: String::from("Cookie could not be set"),
+    });
+
+  r
+}
+
+#[cfg(not(feature = "ssr"))]
+pub async fn remove_cookie_wrapper(path: &str) -> Result<(), LemmyAppError> {
+  use leptos::window;
+  use wasm_bindgen::JsCast;
+
+  let r = window()
+    .document()
+    .ok_or(LemmyAppError::APIError {
+      error: String::from("DOM document is None"),
+    })?
+    .dyn_into::<web_sys::HtmlDocument>()
+    .map_err(|_| LemmyAppError::APIError {
+      error: String::from("DOM document could not be cast"),
+    })?
+    .set_cookie(&wasm_cookies::cookies::delete(path))
+    .map_err(|_| LemmyAppError::APIError {
+      error: String::from("Cookie could not be set"),
+    });
+
+  r
+}
+
+#[cfg(feature = "ssr")]
+pub async fn set_cookie_wrapper(path: &str, value: &str) -> Result<(), LemmyAppError> {
+  use actix_web::{
+    cookie::{
+      time::{Duration, OffsetDateTime},
+      Cookie,
+    },
+    http::{header, header::HeaderValue},
+  };
+  use leptos::expect_context;
+  use leptos_actix::ResponseOptions;
+
+  let response = expect_context::<ResponseOptions>();
+
+  let mut cookie = Cookie::build(path, value).finish();
+  let mut now = OffsetDateTime::now_utc();
+  now += Duration::weeks(1);
+  cookie.set_expires(now);
+  cookie.set_path("/");
+
+  if let Ok(cookie) = HeaderValue::from_str(&cookie.to_string()) {
+    response.insert_header(header::SET_COOKIE, cookie);
+  }
+
+  Ok(())
+}
+
+#[cfg(feature = "ssr")]
+#[allow(dead_code)]
+pub async fn remove_cookie_wrapper(path: &str) -> Result<(), LemmyAppError> {
+  use actix_web::{
+    cookie::{
+      time::{Duration, OffsetDateTime},
+      Cookie,
+    },
+    http::{header, header::HeaderValue},
+  };
+  use leptos::expect_context;
+  use leptos_actix::ResponseOptions;
+
+  let response = expect_context::<ResponseOptions>();
+
+  let mut cookie = Cookie::build(path, "").finish();
+  let mut now = OffsetDateTime::now_utc();
+  now += Duration::weeks(-1);
+  cookie.set_expires(now);
+  cookie.set_path("/");
+
```

Comment:
Does actix cookies not have removing cookies as part of their API?

Reply:
i can't find it explicitly, can you?

i think this is a standard way of removing cookies but i'm open to alternatives